src/App.tsx

```diff
@@ -17,18 +59,36 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                setSelectedFilter={setSelectedFilter}
+                setQuery={setQuery}
+                query={query}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {!areTodosLoaded ? (
+                <Loader />
+              ) : (
+                <TodoList
+                  todos={todos}
+                  setIsModalShown={setIsModalShown}
+                  setSelectedTodo={setSelectedTodo}
+                  selectedTodo={selectedTodo}
+                />
+              )}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {isModalShown && (
```

Comment:
```suggestion
      {selectedTodo && (
```
You can show modal only in case you have a selected todo and get rid of this state